plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -512,4 +605,19 @@ private static TableIdentifier toIdentifier(SchemaTableName schemaTableName)
     {
         return TableIdentifier.of(schemaTableName.getSchemaName(), schemaTableName.getTableName());
     }
+
+    private List<Namespace> relevantNamespaces(ConnectorSession session, Optional<String> namespace)
+    {
+        List<Namespace> namespaces;
+        if (namespace.isPresent() && namespaceExists(session, namespace.get())) {
+            namespaces = ImmutableList.of(Namespace.of(namespace.get()));
+        }
+        else {
+            namespaces = listNamespaces(session).stream()
```

Comment:
No point calling `listNamespaces` when namespace name is given, but namespace of given name does not exist.

Since this is pre-existing bug, fixing this should be a prep commit.